lms/djangoapps/student_account/talenetic.py

```diff
@@ -0,0 +1,220 @@
+from six.moves.urllib_parse import urlencode, unquote
+import jwt
+import json
+from django.conf import settings
+from student.models import Registration, UserProfile
+from social_core.backends.oauth import BaseOAuth2
+from django.contrib.auth.models import User
+import uuid
+import logging
+import social_django
+
+log = logging.getLogger(__name__)
+
+class TaleneticOAuth2(BaseOAuth2):
+    """
+    Talenetic OAuth2 authentication backend
+
+    """
+    settings_dict = settings.CUSTOM_BACKENDS.get('talenetic')
+    name = 'talenetic-oauth2'
+    REDIRECT_STATE = False
+    ID_KEY = 'emailaddress'
+    STATE_PARAMETER = False
+    AUTHORIZATION_URL = settings_dict.get('AUTH_URL')
+    ACCESS_TOKEN_URL = settings_dict.get('ACCESS_TOKEN_URL')
+    ACCESS_TOKEN_METHOD = 'GET'
+    REFRESH_TOKEN_URL = settings_dict.get('REFRESH_TOKEN_URL')
+    REFRESH_TOKEN_METHOD = 'POST'
+    RESPONSE_TYPE = 'code jwt_token'
+    REDIRECT_IS_HTTPS = False
+    REVOKE_TOKEN_URL = settings_dict.get('LOGOUT_URL')
+    REVOKE_TOKEN_METHOD = 'POST'
+
+    def get_scope_argument(self):
+        return {}
+
+
+    def auth_complete(self, *args, **kwargs):
+        """Completes login process, must return user instance"""
+        self.process_error(self.data)
+        state = self.validate_state()
+        access_url = "{}?uid={}".format(self.access_token_url(), self._get_uid())
+        response = self.request_access_token(
+            access_url,
+            data=self._get_creds(),
+            headers=self._get_creds(),
+            auth=self.auth_complete_credentials(),
+            method=self.ACCESS_TOKEN_METHOD
+        )
+        self.process_error(response)
+        return self.do_auth(response['jwt_token'], response=response,
+                            *args, **kwargs)
+
+
+    def do_auth(self, jwt_token, *args, **kwargs):
+        data = self.user_data(jwt_token, *args, **kwargs)
+        response =  kwargs.get('response') or {}
+        response.update(data or {})
+        if 'access_token' not in response:
+            response['access_token'] = jwt_token
+        kwargs.update({'response': response, 'backend': self})
+        return self.strategy.authenticate(*args, **kwargs)
+
+
+    def _get_uid(self):
+        if 'uid' in self.data:
+            return self.data['uid']
+        else:
+            return None
+
+
+    def auth_params(self, state=None):
+        client_id, client_secret = self.get_key_and_secret()

```

Comment:
Where do you get `get_key_and_secret`?

Reply:
@jagonzalr they are in the base class that is being overridden, its a method in the super.